### PR TITLE
Feature verify location

### PR DESCRIPTION
**Pull request by bwan2**
Dependency [PR#17](https://github.com/jian-en/pinocchioNodeServer/pull/17)

### Completed
- Created a new endpoint `api/events/verifyLocation` that takes in an `eventsId`, `latitude`, and `longitude`. The endpoint will check if the user given coordinates are within 0.3 km of the event lat/long. This will return `200` if within 0.3km and `422` if not. 
- Creating an event will add `latitude` and `longitude` to the table.

### Requires
- `node-geocoder` package. This uses `opencage` as the geocoder provider and you must create an API key for the geocoder. Ask Brian for key or create your own here https://opencagedata.com
- copy the `config.js.sample` to get the new `config.js` format

### CODE REVIEW

**Comment by miajhaochen** [comment]
it's possible that the city may be empty 
image
line 92: `const locationCity = locations[i].city.toLowerCase();`

the `locations` returned by the gps service: 
image


(the address I typed in was `230N Craig Street`. Although it was an invalid address, it returns me the location with city `undefined`)

File: src/routes/events.router.js [hunk 1]
```diff
@@ -83,4 +83,39 @@ module.exports = (app) => {
...
+     *      tags:
+     *          - Events
+     *      name: verify location
+     *      summary: Check if the user is in the event location
+     *      produces:
+     *          - application/json
+     *      consumes:
+     *          - application/x-www-form-urlencoded
+     *      parameters:
+     *          - name: eventId
```
Comment (miajhaochen) [inline src/routes/events.router.js]: its says `eventId` but actually you use `eventsId` in code

**Review by miajhaochen — CHANGES_REQUESTED** [review]

File: src/routes/events.router.js (hunk 1 above)
Comment (bwan2) [reply src/routes/events.router.js]: xD

**Comment by bwan2** [comment]
> it's possible that the city may be empty
> image
> line 92: `const locationCity = locations[i].city.toLowerCase();`
> 
> the `locations` returned by the gps service:
> image
> 
> (the address I typed in was `230N Craig Street`. Although it was an invalid address, it returns me the location with city `undefined`)

good catch, i need to add unit tests for this. I fixed the error though

**Comment by bwan2** [comment]
added tests/cleaned some tests @miajhaochen please check again

Note: sometimes geocoder will give undefined back even with a valid location. This is a known issue and will need to be resolved.

**Comment by miajhaochen** [comment]
conflictedddddd

**Pull request merged**